src/main/webapp/app/exercises/shared/exercise-hint/manage/exercise-hint.service.ts

```diff
@@ -51,10 +53,11 @@ export class ExerciseHintService implements IExerciseHintService {
      */
     create(exerciseHint: ExerciseHint): Observable<ExerciseHintResponse> {
         exerciseHint.exercise = ExerciseService.convertDateFromClient(exerciseHint.exercise!);
+        exerciseHint.type = 'text';
         if (exerciseHint.exercise.categories) {
             exerciseHint.exercise.categories = ExerciseService.stringifyExerciseCategories(exerciseHint.exercise);
         }
-        return this.http.post<ExerciseHint>(this.resourceUrl, exerciseHint, { observe: 'response' });
+        return this.http.post<ExerciseHint>(`${this.resourceUrl}/${exerciseHint.exercise.id}/exercise-hints`, exerciseHint, { observe: 'response' });
```

Comment:
I think, similar to the server endpoints, you should take the id, not from the exercise of the hint but pass the query parameter along

Reply:
Please change this for the other occurrences in this file too then